### ISSUE TITLE
App icon refresh?

### ISSUE THREAD

**Issue by bertob** [mixxxdj/mixxx#12863]
### Bug Description

Hi, I just looked at this app from the Flathub metadata review side, and wanted to reach out since the app is now verified. We'll soon roll out curated banners on the Flathub home page and would love to feature Mixxx as part of that, but it doesn't quite meet all the [metadata guidelines](https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/quality-guidelines/) yet.

In addition to the screenshot size being too large (which should be easy to fix), the major issue with this app's listing is the mid-2000s Tango icon style, which looks a bit dated nowadays. Would you be interested in help with modernizing it a bit?

### Version

2.4.0 from Flathub

### OS

Fedora 39

**Comment by ronso0** [comment]
> is the mid-2000s Tango icon style, which looks a bit dated nowadays

Would you mind being a bit more specific?
How is the style of a (not default) theme relevant (major issue?) for listing an app?
How do you feel about Shade :p

**Comment by ronso0** [comment]
or are you referring to blurred / pixelated icons when using a scale factor != 1.0?

**Comment by jimmac** [comment]
The style of the icon follows the Tango guidelines that were set in place a good decade ago. The nominal size was 48x48px and the contrast against any background was guaranteed by using two strokes (darker outer) and inner "highlight" one. The app icon these days is presented at a much bigger size, making those strokes pretty giant and changing the overall feel of the icon.

![image](https://github.com/mixxxdj/mixxx/assets/14821/43e04536-2d4e-44a5-99ef-e1d46841fada)

I think an update of the style is due, as the icon looks out of place not only on free desktop environments, but on all platforms Mixxx ships on. I think the crossfader metaphor is sound and it would be great to keep it, but follow a more [contemporary style](https://developer.gnome.org/hig/guidelines/app-icons.html) (In key elements not too far off from [KDE](https://develop.kde.org/hig/style/icons/colorful/)).

![Mixxx](https://github.com/mixxxdj/mixxx/assets/14821/293d9fc4-ade6-4244-841e-b4c5d07fc8de)

**Comment by ronso0** [comment]
Ah, okay, it's about the Mixxx icon, nit the icons of the Tango skin 😄

**Comment by bertob** [comment]
Here's a quick first attempt at converting the existing icon to a more modern style:

![image](https://github.com/mixxxdj/mixxx/assets/1908896/291c45df-4607-4071-85d3-85e7dbe45a2a)

**Comment by Holzhaus** [comment]
Wow, looks nice. Could you upload the SVG, so that we can play around with it? I'd love to play around and see how it looks with a darker surface under the crossfader.

**Comment by ronso0** [comment]
yeah, this is indeed an improvement!
jup, the notch could be darker.

**Comment by ywwg** [comment]
I like it! I was just thinking that the icon was looking out of date, so the timing of this discussion is perfect.  I am also tempted to see what a totally different modernized icon would look like.  The colors don't really match how the application itself looks -- it's a dark-mode app, but the icon light-colored.  I am happy to merge this new icon as-is, but if people want to get creative and brainstorm a totally new direction, I'd love to see it!

**Comment by ywwg** [comment]
(I realize I am tempting the bike-shed monster haha)

**Comment by ronso0** [comment]
Keep in mind that (Linux) icon themes do override this, so I'm not sure if developing a darkish icon is really worth the effort.
https://github.com/mixxxdj/mixxx/pull/12424#issuecomment-1855519113

**Comment by fwcd** [comment]
If we update the icon, keep in mind that we also need a rectangular version for macOS/iOS:

- https://github.com/mixxxdj/mixxx/blob/main/res/images/icons/scalable/apps/mixxx_macos.svg
- https://github.com/mixxxdj/mixxx/blob/main/res/images/icons/scalable/apps/mixxx_ios.svg

**Comment by Be-ing** [comment]
I think it's awfully inappropriate for Flathub to try to dictate aesthetic choices. I for one don't appreciate people who have had zero involvement with Mixxx coming in and saying we should redesign the icon because of arbitrary reasons you made up. "Looks a bit dated" is irrelevant. "Looks out of place not only on free desktop environments" is subjective, and I disagree:
![image](https://github.com/mixxxdj/mixxx/assets/9455094/a063c39e-1c01-409f-93f8-46ac0e657b8b)

**Comment by Holzhaus** [comment]
@Be-ing i disagree with your assessment of the situation, at least it does not match my perception. 

@bertob stated icon looks dated and asked if we'd be interested in help with modernizing it a bit. This alone would have already sufficed to convince me, because the icon indeed has this GNOME 2.x look and nice that he offered help to improve it. If the redesign also helps to increase our changes to get featured on flathub, that's a nice side effect but hardly carries enough weight to let Flathub "try to dictate aesthetic choices".

> A submission to Flathub is not an invitation for design critique. I for one don't appreciate people who have had zero involvement with Mixxx coming in and saying we should redesign the icon because of arbitrary reasons you made up.

We have people coming into our bug tracker all the time asking for some feature, redesign or change. And this is generally fine IMHO. I don't think you have to "earn" the right to make a suggestion and open a bug report, especially when you also offer help fixing it.

**Comment by Be-ing** [comment]
I disagree that the icon "looks dated", or that if it did, that would matter at all. I don't think chasing fashion trends is a good reason to change established, recognizable branding.

I don't appreciate @bertob's presence here at all and I don't perceive it as helping.

> I don't think you have to "earn" the right to make a suggestion and open a bug report

No, you don't have to earn that right, but when you come in as an outsider and say "you have to do this to meet guidelines I made up", it's a different situation.

**Comment by fwcd** [comment]
> I don't think chasing fashion trends is a good reason to change established, recognizable branding.

Well, if the rest of the world changes, not adapting will necessarily mean that we'll look dated at some point (without judging whether that's good or bad). I would argue that @bertob and @jimmac have made very reasonable attempts at preserving the look and feel of our icon while moving it more in line with a more modern style. They aren't complaining, they are actively contributing towards finding a solution that fits in, which I think is something we should highly appreciate.

> No, you don't have to earn that right, but when you come in as an outsider and say "you have to do this to meet guidelines I made up", it's a different situation.

They are certainly not the only ones doing that:

- https://developer.gnome.org/hig/guidelines/app-icons.html
- https://develop.kde.org/hig/style/icons/colorful/application/
- https://developer.apple.com/design/human-interface-guidelines/app-icons#macOS

**Comment by Be-ing** [comment]
> They are certainly not the only ones doing that:

Those guidelines apply to their own projects and that's fine. Imposing them on unrelated projects is inappropriate.

**Comment by fwcd** [comment]
No, this is exactly what these guidelines are intended for, providing a style guide for apps on their platform (Linux, macOS and Windows), which just so happens to be exactly the platforms we target.

I would even argue that these style guides are pretty similar in the design language they encourage, being roughly centered around the flat design that has been widely adopted across ecosystems for the past decade, so it shouldn't be that hard to come up with something that conforms to all of them (and I would argue the existing suggestions in this thread come pretty close).

**Comment by mxmilkiib** [comment]
New icon n style guide; cool, sure, fine.

Requiring that as a barrier to promotion; to frame it in a not-just-Mixxx way: it feels like the Flatpak world will be a (needless on a practical level) filter for all GUI applications, i.e.,  if a fairly stable and handy app that doesn't see much development but has worth and can readily be packaged as a Flatpak, it still needs a fresh icon to get deigned to be given the opportunity to see that extra boost and respect and attention that living on the front page for a time provides, however long a time goes by? (Through a style guide, darkly.)

**Comment by bragefuglseth** [comment]
> doing the work at the Flatpak end to make and use a custom icon, like a toolkit theme, for promotion, then submitting it as a proposal to Mixxx for a possibly more universal use

Changing the visual identity of an app behind its back, and prominently displaying that, is something I would consider way more disrespectful than what has been done here. 🙂

**Comment by mxmilkiib** [comment]
And not promote the app. It's a very Apple feeling approach, not entirely in the spirit of free software. But Flatpak should do Flatpak, that's not my world.

**Comment by Be-ing** [comment]
I put Mixxx on Flathub to make it easy for users on any Linux distro to install it and keep it updated. That's it. Flathub appointing itself to a role of gatekeeping design is uncalled for.

**Comment by razzeee** [comment]
> And not promote the app. It's a very Apple feeling approach, not entirely in the spirit of free software. But Flatpak should do Flatpak, that's not my world.

Don't conflate flatpak with flathub.

In general the index space (especially above the fold) on flathub.org is a scarce resource and showing apps that put the best foot of the ecosystem forth is just a no brainer.

**Comment by bragefuglseth** [comment]
I can understand how this issue might have come off as pushy or condescending towards the Mixxx community, and seemed like a hostile attempt to override the contributors solely for the sake of how the app looks on Flathub. If that was the case, however, what @mxmilkiib suggested would probably have been done instead, and that's not happening. *That* would have been incredibly disrespectful towards the community. :sweat_smile: As stated, the final decision on this is to be made by the contributors, and it has no impact on the mere availability of the app on Flathub whatsoever.

(As a sidenote, you might be interested in knowing that @jimmac, who drafted the new icon suggestion further up in this thread, is one of the [co-creators of Tango](https://actsofvolition.com/2005/10/announcingthe/), the style used by the current Mixxx app icon! How the tables have turned since back then :slightly_smiling_face:)

The current atmosphere of this conversation makes me think that this issue should be closed for now. There's nothing keeping Mixxx from simply existing on Flathub in its current state anyways, and while I'd personally love to discuss this more, I fear that continuing this right now will just be draining for everybody who participates. We could always pick up the thread on this later, should Mixxx ever be interested.

**Comment by Be-ing** [comment]
> The style of the icon follows the Tango guidelines that were set in place a good decade ago.

FWIW, that may be a coincidence. I don't know whether or not the present icon was designed with any intention of following the Tango guidelines.

**Comment by fwcd** [comment]
> I don't know whether or not the present icon was designed with any intention of following the Tango guidelines.

Interestingly, that seems to be the case: https://github.com/mixxxdj/mixxx/commit/c3deb1678c72a4db66944e27b8a0bf2f8dfa4af4

**Comment by Be-ing** [comment]
This could have, and still could, go very differently if you stepped down from your role of self-appointed gatekeeper of the design of all Linux applications' icons. If you said "hey, we're some designers who want to help application developers improve their icons, would you like some suggestions?" *with no strings attached*, that would go much better.

**Comment by ywwg** [comment]
Because this discussion has gotten contentious, the core team is going to engage its decision process to resolve the dispute. In order to give feelings a chance to settle down, please do not continue the conversation for now. I am going to *temporarily* lock the conversation, but we will reopen it once the process has reached a conclusion, probably in about a week.

First, the core team are going to hold a vote on whether we want to up the app icon. Second, if we do decide to change it, we will decide on a process to pick a new icon, whether that's by an open submission + voting process or some other mechanism. That will provide a predictable path towards making this important change in a way that will reflect the consensus of the project.
